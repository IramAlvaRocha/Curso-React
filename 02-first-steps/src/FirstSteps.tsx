import { ItemCounter } from "./shopping-cart/ItemCounter";

interface ItemInCart {
  productName: string;
  quantity: number;
}

const itemsInCart: ItemInCart[] = [
  { productName: "Nintendo Switch 2", quantity: 3 },
  { productName: "Pro Controller", quantity: 2 },
  { productName: "PlayStation 5 Pro", quantity: 1 },
  { productName: "Super Mario Maker 3", quantity: 4 },
];

export default function FirstSteps() {
  return (
    <>
      <h1>Carrito de compras</h1>

      {itemsInCart.map( ({productName, quantity}) => (
        <ItemCounter 
          key={productName}
          name={productName} 
          quantity={quantity}
          />
      ))}

      {/* <ItemCounter name="Nintendo Switch 2" quantity={3}/>
      <ItemCounter name="Xbox Series X" quantity={2}/>
      <ItemCounter name="PlayStation 5 Pro" quantity={1}/>
      <ItemCounter name="Super Mario Maker 3" quantity={4}/> */}
    </>
  );
}