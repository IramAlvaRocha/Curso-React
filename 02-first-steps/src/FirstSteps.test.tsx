import { render, screen } from "@testing-library/react";
import { afterEach, describe, expect, test, vi } from "vitest";
import FirstSteps from "./FirstSteps";

const mockItemCounter = vi.fn( (_props: unknown) => {
    return( <div data-testid="ItemCounter" />)
} )


vi.mock('./shopping-cart/ItemCounter', ()=>({
    ItemCounter: (props: unknown)=> mockItemCounter(props)
}))
    
    
describe("FirstSteps", () => {

    afterEach(()=>{
        vi.clearAllMocks()
    })

    test('Should match snapshots', ()=>{
        const {container} = render(<FirstSteps />);

        expect(container).matchSnapshot();
    })

    test('Should render the correct number of ItemCounter componentes', ()=>{
        render(<FirstSteps />)

        const counterComponents = screen.getAllByTestId('ItemCounter'); 
        
        expect(counterComponents.length).toBe(4);
        
        screen.debug()
    });

    test('should render ItemCounter with correct props', ()=>{
        render(<FirstSteps />);

        expect(mockItemCounter).toHaveBeenCalledTimes(4);

        expect(mockItemCounter).toHaveBeenCalledWith({ 
            name: "Nintendo Switch 2", 
            quantity: 3, 
        });
        expect(mockItemCounter).toHaveBeenCalledWith({ 
            name: "Pro Controller", 
            quantity: 2, 
        });
        expect(mockItemCounter).toHaveBeenCalledWith({ 
            name: "PlayStation 5 Pro", 
            quantity: 1, 
        });
        expect(mockItemCounter).toHaveBeenCalledWith({ 
            name: "Super Mario Maker 3", 
            quantity: 4, 
        });
    })
})