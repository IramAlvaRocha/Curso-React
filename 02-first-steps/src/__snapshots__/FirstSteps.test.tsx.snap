// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstSteps > Should match snapshots 1`] = `
<div>
  <h1>
    Carrito de compras
  </h1>
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
</div>
`;
